templates/app/CMakeLists.txt.tmpl

```diff
@@ -1,4 +1,8 @@
 cmake_minimum_required(VERSION 3.15)
+
+set(FLUTTER_SYS_ROOT ${CMAKE_SYSROOT})
+set(CMAKE_SYSROOT "")
```

Comment:
Is it necessary to write before the "project"? Is there information about this somewhere?

Reply:
Basically cmake runs it's first compiler tests when we call the [`project` function](https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#method-3-avoid-use-set), this is why we need to unset CMAKE_SYSROOT before it to avoid it using the sysroots make file for it's test compile.

Reply:
One other possible solution is that we can either also set the [CMAKE_SYSROOT_COMPILE](https://cmake.org/cmake/help/latest/variable/CMAKE_SYSROOT_COMPILE.html#variable:CMAKE_SYSROOT_COMPILE) & [CMAKE_SYSROOT_LINK](https://cmake.org/cmake/help/latest/variable/CMAKE_SYSROOT_LINK.html#variable:CMAKE_SYSROOT_LINK) variables separately so cmake will avoid using the make file in our sysroot. However I am unsure if this would work and of why is cmake using make from the sysroot.